hamt/hamt.go

```diff
@@ -419,75 +434,109 @@ func (ds *Shard) walkTrie(ctx context.Context, cb func(*Shard) error) error {
 	})
 }
 
-func (ds *Shard) modifyValue(ctx context.Context, hv *hashBits, key string, val *ipld.Link) error {
+// setValue sets the link `value` in the given key, either creating the entry
+// if it didn't exist or overwriting the old one. It returns the old entry (if any).
+func (ds *Shard) setValue(ctx context.Context, hv *hashBits, key string, value *ipld.Link) (oldValue *ipld.Link, err error) {
 	idx, err := hv.Next(ds.tableSizeLg2)
 	if err != nil {
-		return err
+		return
```

Comment:
nit: I find it much easier to reason about longer functions like these to be explicit about the return types instead of using `return` and predeclared return variables.

Reply:
I'm a bit on the fence on this one but will fix in this function if it helps readability (as you say it's a rather long one).

Reply:
https://github.com/ipfs/go-unixfs/issues/105